paymentsheet/src/main/java/com/stripe/android/customersheet/StripeCustomerAdapter.kt

```diff
@@ -113,15 +126,15 @@ internal class StripeCustomerAdapter @Inject constructor(
         }
     }
 
-    override suspend fun setupIntentClientSecretForCustomerAttach(): Result<String> {
+    override suspend fun setupIntentClientSecretForCustomerAttach(): CustomerAdapterResult<String> {
         return getCustomerEphemeralKey().mapCatching { customerEphemeralKey ->
             setupIntentClientSecretProvider?.provide(customerEphemeralKey.customerId)
-        }.getOrElse {
-            Result.failure(it)
+        }.getOrElse { cause ->
+            CustomerAdapterResult.Failure(cause, cause.message)
```

Comment:
I think we should pass null, rather than the exception message.

Reply:
It isn't clear in this, but the exception is coming from `CustomerAdapterResult.Failure`